jdaviz/configs/imviz/plugins/links_control/links_control.py

```diff
@@ -214,16 +215,12 @@ def _update_link(self, msg={}):
             is_wcs_only = data.meta.get(self.app._wcs_only_label, False)
             has_wcs = hasattr(data.coords, 'pixel_to_world')
             if not is_wcs_only:
-                if data not in data_in_viewer:
-                    # add data from viewer:
-                    if wcs_linked and has_wcs:
-                        self.app.add_data_to_viewer(viewer_selected.reference, data.label)
-                    elif not wcs_linked:
-                        self.app.add_data_to_viewer(viewer_selected.reference, data.label)
-
-                elif wcs_linked and not has_wcs:
+                if data in data_in_viewer and wcs_linked and not has_wcs:

```

Comment:
@pllim - we'll also need to make sure that this logic works for those tables so they are not automatically removed from the viewer when changing linking type.

We really should consolidate the "does this data support plotting on the sky" logic in a single place, if anyone has suggestions, we could do that in this PR, otherwise defer as tech debt.